store/src/java/com/zimbra/cs/mailbox/MailSender.java

```diff
@@ -1015,7 +1003,12 @@ void updateHeaders(MimeMessage mm, Account acct, Account authuser, OperationCont
         mm.setFrom(from);
         mm.setSender(sender);
 
-        mm.setSentDate(new Date());
+        Date date = new Date();
+        mm.setSentDate(date);
+
+        String value = SecretKey.getMessageVerificationHeaderValue(mm.getMessageID(), mailDateFormat.format(date), from.getAddress());
```

Comment:
Use java 8 practices
` Optional.ofNullable(SecretKey.getMessageVerificationHeaderValue(
        mm.getMessageID(),
        mailDateFormat.format(date),
        from.getAddress()
)).ifPresent(value -> mm.addHeader(X_MESSAGE_VERIFICATION, value));
`

Reply:
I tried that but encountered an implementation issue because the lambda expression used in Optional.ofNullable(...).ifPresent(...) cannot handle checked exceptions like MessagingException. This is a limitation of Java's lambda expressions, which don't natively support checked exceptions.

Reply:
done